Mathlib/NumberTheory/NumberField/Norm.lean

```diff
@@ -43,60 +43,67 @@ namespace RingOfIntegers
 variable {L : Type*} (K : Type*) [Field K] [Field L] [Algebra K L] [FiniteDimensional K L]
 
 /-- `Algebra.norm` as a morphism betwen the rings of integers. -/
-@[simps!]
 noncomputable def norm [IsSeparable K L] : 𝓞 L →* 𝓞 K :=
-  ((Algebra.norm K).restrict (𝓞 L)).codRestrict (𝓞 K) fun x => isIntegral_norm K x.2
+  RingOfIntegers.restrict_monoidHom
```

Comment:
I know this is orthogonal to the PR but does this really need separability?

Reply:
I think it holds in general, but this is not completely clear. Maybe one has to break the extension in separable and purely inseparable parts, and see what happens in the purely inseparable case. Anyway this is meant to be used for number fields...